packages/strapi-design-system/src/TimePicker/TimePicker.tsx

```diff
@@ -1,77 +1,82 @@
 import { useId } from '../hooks/useId';
-import { SingleSelect, SingleSelectOption, SingleSelectProps } from '../Select/SingleSelect';
 
-export interface TimePickerProps extends Omit<SingleSelectProps, 'children' | 'onChange' | 'value'> {
-  onChange: (value: string) => void;
+/* -------------------------------------------------------------------------------------------------
+ * TimePickerInput
+ * -----------------------------------------------------------------------------------------------*/
+
+export interface TimePickerInputProps
+  extends Omit<ComboboxInputProps, 'children' | 'autocomplete' | 'startIcon' | 'placeholder' | 'allowCustomValue'> {
   /**
    * @default 15
    */
   step?: number;
-  value?: string;
+  /**
+   * @deprecated This is no longer used.
+   */
+  ariaLabel?: string;
+  /**
+   * @preserve
+   * @deprecated This is no longer used.
+   */
+  selectButtonTitle?: string;
 }
 
-export const TimePicker = ({ id, value, step = 15, onChange, ...props }: TimePickerProps) => {
+export const TimePickerInput = ({
+  id,
+  step = 15,
+  /**
+   * @preserve
+   * @deprecated This is no longer used.
+   */
+  ariaLabel: _ariaLabel,
+  /**
+   * @preserve
+   * @deprecated This is no longer used.
+   */
+  selectButtonTitle: _selectButtonTitle,
+  ...restProps
+}: TimePickerInputProps) => {
+  const context = useDesignSystem('TimePicker');
   const generatedId = useId(id);
-  const hoursCount = 24;
-  const times: string[] = [];
-  let min = 0;
-
-  for (let i = 0; i < hoursCount; i++) {
-    min = 0;
-
-    while (min < 60) {
-      times.push(`${i < 10 ? `0${i}` : i}:${min < 10 ? `0${min}` : min}`);
-      min += step;
-    }
-  }
-
-  // The time picker will select the closest value in the list.
-  // This is a temporary fix.
-  // This whole thing needs refactoring – it's nonsensical.
-  const getClosestValue = () => {
-    const [valueHours, valueMinutes] = value?.split(':') ?? [];
 
-    const hours = times.reduce((prev, curr) => {
-      const [h] = curr.split(':');
+  const formatter = useDateFormatter(context.locale, {
+    hour: '2-digit',
+    minute: '2-digit',
+  });
 
-      // @ts-expect-error this is gonna be refactored in an upcoming initiative
-      return Math.abs(h - valueHours) < Math.abs(prev - valueHours) ? h : prev;
-    }, times[0].split(':')[0]);
+  const timeOptions = React.useMemo(() => {
+    const stepCount = 60 / step;
 
-    const minutes = times.reduce((prev, curr) => {
-      const minutes = curr.split(':')[1];
-
-      // @ts-expect-error this is gonna be refactored in an upcoming initiative
-      return Math.abs(minutes - valueMinutes) < Math.abs(prev - valueMinutes) ? minutes : prev;
-    }, times[0].split(':')[1]);
-
-    return `${hours}:${minutes}`;
-  };
-
-  const handleChange = (value: string | number | string[]) => {
-    if (onChange) {
-      onChange(value.toString());
-    }
-  };
+    return [...Array(24).keys()].flatMap((hour) =>
```

Comment:
What if the user manually introduces 99:99, shouldn't it trigger a validation error? Maybe it's part of the outside logic that uses this component? What do you think?

Reply:
Excellent question, in another PR we're gonna work on validation logic ✌🏼